wallet/src/actors/app/methods.rs

```diff
@@ -371,11 +371,19 @@ impl App {
                     || slf.state.node_state == None
                 {
                     let sink = slf.state.get_sink(&session_id);
-                    slf.params.worker.do_send(worker::SyncRequest {
-                        wallet_id,
-                        wallet,
-                        sink,
-                    });
+                    slf.params
+                        .worker
+                        .send(worker::SyncRequest {
+                            wallet_id,
+                            wallet,
+                            sink,
+                        })
+                        .flatten()
+                        .into_actor(slf)
+                        .map_err(|e, act: &mut Self, _ctx| {
+                            act.handle_sync_error(&(e.into()));
+                        })
+                        .wait(ctx)
```

Comment:
I think this `wait` will make the `App` actor to stop processing any messages until the synchronization is done (or there is a timeout). So maybe this breaks the ability to generate addresses while the wallet is syncing. Not sure, I will have to try.

Reply:
interesting...I can also trry this

Reply:
It looks like this will result in a deadlock if you try to unlock a different wallet while the first wallet is syncing... So let's try to change the `wait` into a `spawn`.

Reply:
Using `spawn` seems to work as expected.

Reply:
good